### PR TITLE
feat(menu): Menu auto-positioning based on an anchor.

### PR DESCRIPTION
**Pull request by sgomes**
Adds support for auto-positioning based on an anchor, with the menu
automatically selecting which corner to open from, based on defaults
and the presence of close screen edges.

@traviskaufman PTAL!

### CODE REVIEW

File: packages/mdl-menu/simple/foundation.js [hunk 1]
```diff
@@ -299,13 +306,57 @@ export default class MDLSimpleMenuFoundation extends MDLFoundation {
...
+
+    // Defaults: open from the top left.
+    let vertical = 'top';
+    let horizontal = 'left';
+
+    const anchor = this.adapter_.getAnchorDimensions();
+    const windowDimensions = this.adapter_.getWindowDimensions();
+
+    // Would the preferred position go beyond the vertical bounds of the window?
+    const topOverflow = anchor.top + this.dimensions_.height - windowDimensions.height;
```
Comment (traviskaufman) [inline packages/mdl-menu/simple/foundation.js]: nit: maybe turn these comments into boolean vars?

```js
const topOverflow = // ...;
const extendsBeyondVerticalBounds = topOverflow > 0;
if (extendsBeyondVerticalBounds) {
  const shouldOpenFromBottom = // ...;
  if (shouldOpenFromBottom) {
    vertical = 'bottom';
  }
}
```

File: packages/mdl-menu/simple/foundation.js [hunk 2]
```diff
@@ -299,13 +306,57 @@ export default class MDLSimpleMenuFoundation extends MDLFoundation {
...
+        vertical = 'bottom';
+      }
+    }
+
+    const leftOverflow = anchor.left + this.dimensions_.width - windowDimensions.width;
+    const rightOverflow = this.dimensions_.width - anchor.right;
+    if (this.adapter_.isRtl()) {
+      // In RTL, we prefer to open from the right.
+      horizontal = 'right';
+      // Would the preferred position go beyond the horizontal bounds of the window?
```
Comment (traviskaufman) [inline packages/mdl-menu/simple/foundation.js]: nit: same as above

File: packages/mdl-menu/simple/foundation.js [hunk 3]
```diff
@@ -299,13 +306,57 @@ export default class MDLSimpleMenuFoundation extends MDLFoundation {
...
+      // If so, would opening from the left be better?
+      if (rightOverflow > 0 && leftOverflow < rightOverflow) {
+        horizontal = 'left';
+      }
+    } else if (leftOverflow > 0 && rightOverflow < leftOverflow) {
+      horizontal = 'right';
+    }
+
+    const position = {};
```
Comment (traviskaufman) [inline packages/mdl-menu/simple/foundation.js]: I believe you can use computed property names here

```js
const position = {
  [horizontal]: '0',
  [vertical]: '0'
};
```

File: packages/mdl-menu/simple/mdl-simple-menu.scss [hunk 4]
```diff
@@ -102,33 +99,22 @@ $mdl-simple-menu-item-offset: 10px;
...
   }
-  [dir="rtl"] &__items {
-    transform-origin: calc(100% - #{$mdl-simple-menu-item-offset}) $mdl-simple-menu-item-offset;
-  }
   /* stylelint-enable plugin/selector-bem-pattern */
 
-  &--open-from-top-right {
-    transform-origin: top right;
+  &--open-from-top-left {
+    transform-origin: top left !important;
```
Comment (traviskaufman) [inline packages/mdl-menu/simple/mdl-simple-menu.scss]: should we make a note in regards to why `!important` is needed here?

File: packages/mdl-menu/simple/foundation.js (hunk 1 above)
Comment (sgomes) [inline packages/mdl-menu/simple/foundation.js]: I could do that, but then I would lose the ability to determine for very large menus which option is "less bad" in terms of overflow (see e.g. line 325, 337 or 340).

File: packages/mdl-menu/simple/foundation.js (hunk 3 above)
Comment (sgomes) [inline packages/mdl-menu/simple/foundation.js]: Nice! I'm still getting used to all the ES2015 stuff :)

File: packages/mdl-menu/simple/mdl-simple-menu.scss (hunk 4 above)
Comment (sgomes) [inline packages/mdl-menu/simple/mdl-simple-menu.scss]: Good idea. Added.

File: packages/mdl-menu/simple/foundation.js (hunk 1 above)
Comment (sgomes) [inline packages/mdl-menu/simple/foundation.js]: Oh wait, duh. Totally misread that. Rewriting.

File: packages/mdl-menu/simple/foundation.js (hunk 2 above)
Comment (sgomes) [inline packages/mdl-menu/simple/foundation.js]: Rewritten too.

**Pull request merged**